knowledge-base/grid-edit-on-row-double-click.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Edit a row on click or double-click with a popup form
+description: How to edit a row on click or double-click with a popup form
+type: how-to
+page_title: Edit a row on click or double-click with a popup form
+slug: grid-edit-row-click
+position: 
+tags: 
+ticketid: 1443681
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>Grid for Blazor</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+
+I would like to edit a row in the Grid when the user clicks or double-clicks on it, instead of using the command buttons.
+
+## Solution
+
+The Grid exposes two events that allows you to respond to the user clicking on its rows - [OnRowClick]({%slug grid-events%}#onrowclick) and [OnRowDoubleClick]({%slug grid-events%}#onrowdoubleclick). You can use either one of them together with the [Window]({%slug components/window/overview%}) to create a custom popup form on a click of a row. 
+
+>caption Use the OnRowDoubleClick event to open a custom popup form to edit the Grid
+
+````CSHTML
+@* Click on a Grid row twice to see the custom edit form *@ 
+
+<TelerikGrid Data="@MyData"
+             Height="400px"
+             Width="700px"
+             Pageable="true"
+             OnRowDoubleClick="@OnRowDoubleClickHandler">
+    <GridColumns>
+        <GridColumn Field="@(nameof(SampleData.Id))" Width="120px" />
+        <GridColumn Field="@(nameof(SampleData.Name))" Title="Employee Name" Groupable="false" />
+        <GridColumn Field="@(nameof(SampleData.Team))" Title="Team" />
+        <GridColumn Field="@(nameof(SampleData.HireDate))" Title="Hire Date" />
+    </GridColumns>
+</TelerikGrid>
+
+<TelerikWindow @bind-Visible="@isInEdit" Modal="true">
+    <WindowTitle>
+        Edit record
+    </WindowTitle>
```

Comment:
May be it is valid to include WindowActions>Close and have a CancelEdit event subscribed to it?

Reply:
The form should be closed by using a `Cancel` button or else depending on the need of the application. The correct way would be to close the popup by using a form button rather than a Window action button.

Reply:
I would just rename the Form button to "Cancel" instead of "Clear". It looks to me more suitable for the current context.